components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/user/OpenIDConnectUserEndpoint.java

```diff
@@ -76,18 +79,26 @@ public Response getUserClaims(@Context HttpServletRequest request) throws OAuthS
                the tenantId in a thread local variable.
              */
             setServiceProviderTenantId(accessToken);
+            OAuthAppDAO oAuthAppDAO = new OAuthAppDAO();
 
             // build the claims
             //ToDO - Validate the grant type to be implicit or authorization_code before retrieving claims
-            UserInfoResponseBuilder userInfoResponseBuilder =
-                    UserInfoEndpointConfig.getInstance().getUserInfoResponseBuilder();
-            response = userInfoResponseBuilder.getResponseString(tokenResponse);
+            String grantType = oAuthAppDAO.getGrantType(accessToken);
+            if (!GrantType.CLIENT_CREDENTIALS.toString().equals(grantType)) {
+                UserInfoResponseBuilder userInfoResponseBuilder =
+                        UserInfoEndpointConfig.getInstance().getUserInfoResponseBuilder();
+                response = userInfoResponseBuilder.getResponseString(tokenResponse);
+            }
 
         } catch (UserInfoEndpointException e) {
             return handleError(e);
         } catch (OAuthSystemException e) {
             log.error("UserInfoEndpoint Failed", e);
             throw new OAuthSystemException("UserInfoEndpoint Failed");
+        } catch (IdentityOAuthAdminException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Error in getting grant type.", e);
+            }
```

Comment:
@DMHP If debug is not enabled we don't get any error details, why ? 

Additionally within the same try-catch block we have used 3 different styles, we have to use one consistent style in all the places.